frontend/src/components/elements/BokehChart/BokehChart.tsx

```diff
@@ -35,6 +38,20 @@ class BokehChart extends React.PureComponent<Props> {
 
   private updateChart = (data: any): void => {
     const chart = document.getElementById(this.chartId)
+    const plot =
+      data && data.doc && data.doc.roots && data.doc.roots.references
+        ? data.doc.roots.references.find((e: any) => e.type === "Plot")
+        : undefined
+    if (plot) {
+      if (!plot.attributes.plot_width) {
+        plot.attributes.plot_width = this.props.width
+      }
+      if (!plot.attributes.plot_height) {
+        plot.attributes.plot_height = this.props.height
+          ? this.props.height
+          : DEFAULT_HEIGHT
+      }
+    }
```

Comment:
It would be nice to move all of the above into a function like: `set_chart_dimensions(...)`

Same with other types of chart, below.

That would keep these `render()` methods readable. Also, those functions can easily be unit-tested.

Reply:
done